programs/whirlpool/src/util/shared.rs

```diff
@@ -75,3 +75,64 @@ pub fn is_locked_position(
 ) -> bool {
     position_token_account.is_frozen()
 }
+
+/// Resolve one-sided position ticks when one of the bounds is auto-derived from the current price.
+///
+/// Rules:
+/// - Exactly one bound may be auto-derived:
+///   - `tick_lower_index == i32::MIN` → lower = ceil(current_tick, tick_spacing)
+///   - `tick_upper_index == i32::MAX` → upper = floor(current_tick, tick_spacing)
+/// - If both sentinels are provided, return `InvalidTickIndex`.
+/// - Ensure `lower < upper` after snapping; otherwise return `InvalidTickIndex`.
+pub fn resolve_one_sided_position_ticks(
+    tick_lower_index: i32,
+    tick_upper_index: i32,
+    tick_spacing: u16,
+    current_tick: i32,
+) -> Result<(i32, i32)> {
+    let mut resolved_tick_lower_index = tick_lower_index;
+    let mut resolved_tick_upper_index = tick_upper_index;
+    let lower_is_sentinel = tick_lower_index == i32::MIN;
+    let upper_is_sentinel = tick_upper_index == i32::MAX;
+    if lower_is_sentinel || upper_is_sentinel {
+        if lower_is_sentinel && upper_is_sentinel {
+            return Err(ErrorCode::InvalidTickIndex.into());
+        }
```

Comment:
we might be able to exit early here in 2 cases:

1. both sentinals are provided -> error
```rust
if lower_is_sentinel && upper_is_sentinel
```
2. neither sentinals are provided -> return early
```rust
if !lower_is_sentinel && !upper_is_sentinel
```

i think this is nice b/c it gets rid of the big surrounding if statement going from 97-136. after the early returns, you know you are working with exactly 1 bound = sentinal value